egs/direction_tasnet/preprocess.py

```diff
@@ -0,0 +1,126 @@
+from torch import nn
+from conv_stft import STFT
+from torch.autograd import Variable
+from deepbeam import R_global
+import numpy as np
+import torch
+sr = 48000
+n_mic = 6
+filter_length = int(sr * 0.004)
+hop_length = int(sr * 0.00125)  # doesn't need to be specified. if not specified, it's the same as filter_length
+window = 'hann'
+win_length = int(sr * 0.0025)
+stft = STFT(filter_length=filter_length, hop_length=hop_length, win_length=win_length,
+            window=window)
+
+
+
+m_total = 97
+frequency_vector = np.linspace(0, sr//2, m_total)
+n_grid = 36
+V = 343
+z = np.zeros((n_mic, n_grid, m_total))
+a = np.arange(m_total).reshape(1, 1, m_total)
+m_data = torch.from_numpy(z + a)
+n_sp = 3
+
+
+
+
+def Prep(data):

```

Comment:
This is where the features are computed right? 
If you ask specific review points, I can read some code, with comments. But otherwise, the intent doesn't look clear enough that it will be fast to review.